### ISSUE TITLE
Detecting if context is db:migrate (or other rake task) depends on executed command 

### ISSUE THREAD

**Issue by dssjoblom** [rails/rails#44735]
### Steps to reproduce

Create an initializer that contains `puts ARGV`.

Run

`bundle exec rake db:migrate`

and 

`rails db:migrate`

For more info on the use case, see https://stackoverflow.com/questions/1858230/how-to-check-if-rails-code-is-running-within-a-migration. 

### Expected behavior

Both should produce the same output (db:migrate), and it has been the behavior earlier in Rails 5.1. Also, if there is an official way of detecting the context it would be nice if it was clearly documented somewhere.

### Actual behavior

In Rails 6.1, the latter command outputs nothing.

### System configuration

**Rails version**: 6.1.4.7 vs. 5.1.7

**Ruby version**: ruby 2.7.5p203

**Comment by HParker** [comment]
Looks like this does not reproduce on Rails 7+

```
$ rails --version
Rails 7.0.2.3

cat db/migrate/20220321162226_test_argv.rb
class TestArgv < ActiveRecord::Migration[7.0]
  def change
    puts ARGV.inspect
  end
end

bundle exec rake db:migrate
== 20220321162226 TestArgv: migrating =========================================
["db:migrate"]
== 20220321162226 TestArgv: migrated (0.0001s) ================================

bundle exec rails db:drop db:create

rails db:migrate
Running via Spring preloader in process 78192
== 20220321162226 TestArgv: migrating =========================================
["db:migrate"]
== 20220321162226 TestArgv: migrated (0.0000s) ================================
```

I don't believe this is the kind of issue we would do a back port for since the current policy is only the latest release receives bug fixes. https://guides.rubyonrails.org/maintenance_policy.html#bug-fixes

**Comment by gmcgibbon** [comment]
@HParker is correct, bug fixes are only accepted for 7.0+ right now. Please upgrade your Rails app! 😄
